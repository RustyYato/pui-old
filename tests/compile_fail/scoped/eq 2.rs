fn main() {
    pui::make_scoped!(a);
    pui::make_scoped!(b);
    assert_eq!(a, b);
}
