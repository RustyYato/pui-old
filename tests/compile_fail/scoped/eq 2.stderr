warning: missing documentation for an associated type
  --> $DIR/runtime.rs:52:5
   |
52 |     type Id: Copy + Eq;
   |     ^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/lib.rs:2:9
   |
2  | #![warn(missing_docs)]
   |         ^^^^^^^^^^^^

warning: 1 warning emitted

error[E0597]: `inv` does not live long enough
 --> $DIR/eq 2.rs:3:5
  |
3 |     pui::make_scoped!(b);
  |     ^^^^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
4 |     assert_eq!(a, b);
5 | }
  | -
  | |
  | `inv` dropped here while still borrowed
  | borrow might be used here, when `_assert_unique_lifetime` is dropped and runs the `Drop` code for type `AssertUniqueLifetime`
  |
  = note: values in a scope are dropped in the opposite order they are defined
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
